src/compartment.c

```diff
@@ -726,16 +728,29 @@ parse_lib_relaplt(Elf64_Shdr *rela_plt_shdr, Elf64_Ehdr *lib_ehdr, int lib_fd,
     char *dyn_str_tbl = malloc(dyn_str_hdr.sh_size);
     do_pread(lib_fd, dyn_str_tbl, dyn_str_hdr.sh_size, dyn_str_hdr.sh_offset);
 
-    lib_dep->rela_maps = malloc(rela_count * sizeof(struct LibRelaMapping));
-    lib_dep->rela_maps_count = rela_count;
+    struct LibRelaMapping *new_relas
+        = malloc(rela_count * sizeof(struct LibRelaMapping));
 
     // Log symbols that will need to be relocated eagerly at maptime
     Elf64_Rela curr_rela;
-    for (size_t j = 0; j < lib_dep->rela_maps_count; ++j)
+    size_t actual_relas = 0;
+    for (size_t j = 0; j < rela_count; ++j)
     {
-        curr_rela = rela_plt[j];
+        curr_rela = rela_sec[j];
         size_t curr_rela_sym_idx = ELF64_R_SYM(curr_rela.r_info);
         Elf64_Sym curr_rela_sym = dyn_sym_tbl[curr_rela_sym_idx];
+
+        // Filter out some `libc` symbols we don't want to handle
+        // TODO at least right now
+        if (!strcmp(&dyn_str_tbl[curr_rela_sym.st_name], "environ"))
+        {
+            continue;
```

Comment:
Should we print warnings for these, or will they fail in some nice determinstic way if someone uses them?

Reply:
It's likely these will crash if used, although I haven't tested it. For now, I've added warnings to avoid using them in https://github.com/capablevms/CHERI-ELF-comp/pull/25/commits/13fcebdc6c36c3671b22d9cb9081b1229e3dd17d.